test/performance/odp_ipsec.c

```diff
@@ -784,20 +784,181 @@ run_measure_one_async(ipsec_args_t *cargs,
 	return rc < 0 ? rc : 0;
 }
 
+#define MAX_ALG_CAPA 32
+
+static int
+ipsec_check(odp_ipsec_capability_t *capa, ipsec_alg_config_t *config,
+	    ipsec_args_t *cargs)
+{
+	odp_ipsec_cipher_capability_t cipher_capa[MAX_ALG_CAPA];
+	odp_ipsec_auth_capability_t   auth_capa[MAX_ALG_CAPA];
+	odp_bool_t found;
+	int i, num;
+
+	if (cargs->ah && (ODP_SUPPORT_NO == capa->proto_ah)) {
+		printf("    AH protocol not supported\n");
+		return -1;
+	}
+
+	/* Cipher algorithms */
+	switch (config->crypto.cipher_alg) {
+	case ODP_CIPHER_ALG_NULL:
+		if (!capa->ciphers.bit.null)
+			return -1;
+		break;
+	case ODP_CIPHER_ALG_DES:
+		if (!capa->ciphers.bit.des)
+			return -1;
+		break;
+	case ODP_CIPHER_ALG_3DES_CBC:
+		if (!capa->ciphers.bit.trides_cbc)
+			return -1;
+		break;
+	case ODP_CIPHER_ALG_AES_CBC:
+		if (!capa->ciphers.bit.aes_cbc)
+			return -1;
+		break;
+	case ODP_CIPHER_ALG_AES_CTR:
+		if (!capa->ciphers.bit.aes_ctr)
+			return -1;
+		break;
+	case ODP_CIPHER_ALG_AES_GCM:
+		if (!capa->ciphers.bit.aes_gcm)
+			return -1;
+		break;
+	case ODP_CIPHER_ALG_AES_CCM:
+		if (!capa->ciphers.bit.aes_ccm)
+			return -1;
+		break;
+	case ODP_CIPHER_ALG_CHACHA20_POLY1305:
+		if (!capa->ciphers.bit.chacha20_poly1305)
+			return -1;
+		break;
+	default:
+		printf("    Unsupported cipher algorithm\n");
+		return -1;
+	}
+
+	/* Authentication algorithms */
+	switch (config->crypto.auth_alg) {
+	case ODP_AUTH_ALG_NULL:
+		if (!capa->auths.bit.null)
+			return -1;
+		break;
+	case ODP_AUTH_ALG_MD5_HMAC:
+		if (!capa->auths.bit.md5_hmac)
+			return -1;
+		break;
+	case ODP_AUTH_ALG_SHA1_HMAC:
+		if (!capa->auths.bit.sha1_hmac)
+			return -1;
+		break;
+	case ODP_AUTH_ALG_SHA256_HMAC:
+		if (!capa->auths.bit.sha256_hmac)
+			return -1;
+		break;
+	case ODP_AUTH_ALG_SHA384_HMAC:
+		if (!capa->auths.bit.sha384_hmac)
+			return -1;
+		break;
+	case ODP_AUTH_ALG_SHA512_HMAC:
+		if (!capa->auths.bit.sha512_hmac)
+			return -1;
+		break;
+	case ODP_AUTH_ALG_AES_XCBC_MAC:
+		if (!capa->auths.bit.aes_xcbc_mac)
+			return -1;
+		break;
+	case ODP_AUTH_ALG_AES_GCM:
+		if (!capa->auths.bit.aes_gcm)
+			return -1;
+		break;
+	case ODP_AUTH_ALG_AES_GMAC:
+		if (!capa->auths.bit.aes_gmac)
+			return -1;
+		break;
+	case ODP_AUTH_ALG_AES_CCM:
+		if (!capa->auths.bit.aes_ccm)
+			return -1;
+		break;
+	case ODP_AUTH_ALG_CHACHA20_POLY1305:
+		if (!capa->auths.bit.chacha20_poly1305)
+			return -1;
+		break;
+	default:
+		printf("    Unsupported authentication algorithm\n");
+		return -1;
+	}
+
+	num = odp_ipsec_cipher_capability(config->crypto.cipher_alg,
+					  cipher_capa, MAX_ALG_CAPA);
+	if (num <= 0) {
+		printf("    Wrong cipher capabilities\n");
```

Comment:
I think this is a bit unclear message, even though the same message is in the validation test. Something like "Could not get cipher capabilities" or "Cipher capability call failed" would be more clear, I think.

Reply:
Agreed. Will update this in the next version.